automation_tools/__init__.py

```diff
@@ -2156,50 +2153,49 @@ def product_install(distribution, certificate_url=None, selinux_mode=None, sat_v
             'custom_repo_{}'.format(k+1): v
             for (k, v) in zip(range(len(custom_repos)), custom_repos)
         }
-        execute(create_custom_repos, host=host, **custom_repos_dict)
+        execute(create_custom_repos, **custom_repos_dict)
     # Update the machine
-    execute(update_packages, host=host, warn_only=True)
+    execute(update_packages, warn_only=True)
 
-    execute(setenforce, selinux_mode, host=host)
+    execute(setenforce, selinux_mode)
 
-    execute(setup_satellite_firewall, host=host)
+    execute(setup_satellite_firewall)
 
-    execute(setup_avahi_discovery, host=host)
+    execute(setup_avahi_discovery)
 
-    execute(run_command, os.environ.get('FIX_PREINSTALL'), host=host)
+    execute(run_command, os.environ.get('FIX_PREINSTALL'))
     # Sat6.3: enable *internal* puppet4 repo to perform fresh p4 install
     if puppet4 == 'yes' and not distribution.endswith('cdn'):
         puppet4_repo = os.environ.get('PUPPET4_REPO')
         if puppet4_repo:
-            execute(create_custom_repos, puppet4_repo=puppet4_repo, host=host)
+            execute(create_custom_repos, puppet4_repo=puppet4_repo)
 
-    # execute returns a dictionary mapping host strings to the given task's
-    # return value
-    installer_options.update(execute(
-        install_tasks[distribution],
-        host=host, run_katello_installer=False
-    )[host])
+    # execute returns a dictionary mapping host strings to the given task's return value
+    installer_options.update(
+        next(iter(execute(install_tasks[distribution], run_katello_installer=False).values()))
```

Comment:
`next(iter(d.values()))` is used in order to get first value of dict (and dict contains always only 1 value) 
it's the same as `list(d.values())[0]`

`d.values()[0]` works only in python2

Reply:
Before the change we got `d['host']` value but by omitting `host=host` we cannot use it, so get just first value `d[0]`